\section{Plugin\+:\+:Plugin\+Exception Class Reference}
\label{class_plugin_1_1_plugin_exception}\index{Plugin\+::\+Plugin\+Exception@{Plugin\+::\+Plugin\+Exception}}


Define the exception used by the \doxyref{Plugin\+Manager}{p.}{class_plugin_1_1_plugin_manager} class.  




{\ttfamily \#include $<$plugin\+\_\+exception.\+h$>$}

Inheritance diagram for Plugin\+:\+:Plugin\+Exception\+:\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[height=2.000000cm]{class_plugin_1_1_plugin_exception}
\end{center}
\end{figure}
\subsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
{\bf Plugin\+Exception} (const char $\ast$message)  throw ()
\begin{DoxyCompactList}\small\item\em Constructor. \end{DoxyCompactList}\item 
virtual {\bf $\sim$\+Plugin\+Exception} (void)  throw ()\label{class_plugin_1_1_plugin_exception_a238039cde46f631071923eefa3822201}

\begin{DoxyCompactList}\small\item\em Destructor. \end{DoxyCompactList}\item 
const char $\ast$ {\bf what} (void) const   throw ()
\begin{DoxyCompactList}\small\item\em Return the error message. \end{DoxyCompactList}\end{DoxyCompactItemize}


\subsection{Detailed Description}
Define the exception used by the \doxyref{Plugin\+Manager}{p.}{class_plugin_1_1_plugin_manager} class. 

\subsection{Constructor \& Destructor Documentation}
\index{Plugin\+::\+Plugin\+Exception@{Plugin\+::\+Plugin\+Exception}!Plugin\+Exception@{Plugin\+Exception}}
\index{Plugin\+Exception@{Plugin\+Exception}!Plugin\+::\+Plugin\+Exception@{Plugin\+::\+Plugin\+Exception}}
\subsubsection[{Plugin\+Exception(const char $\ast$message)}]{\setlength{\rightskip}{0pt plus 5cm}Plugin\+::\+Plugin\+Exception\+::\+Plugin\+Exception (
\begin{DoxyParamCaption}
\item[{const char $\ast$}]{message}
\end{DoxyParamCaption}
) throw  ) }\label{class_plugin_1_1_plugin_exception_ab10ce054f500967a2b1ad78dd8683540}


Constructor. 


\begin{DoxyParams}{Parameters}
{\em message} & \+: Contains the error message. \\
\hline
\end{DoxyParams}


\subsection{Member Function Documentation}
\index{Plugin\+::\+Plugin\+Exception@{Plugin\+::\+Plugin\+Exception}!what@{what}}
\index{what@{what}!Plugin\+::\+Plugin\+Exception@{Plugin\+::\+Plugin\+Exception}}
\subsubsection[{what(void) const }]{\setlength{\rightskip}{0pt plus 5cm}const char $\ast$ Plugin\+::\+Plugin\+Exception\+::what (
\begin{DoxyParamCaption}
\item[{void}]{}
\end{DoxyParamCaption}
) const throw  ) }\label{class_plugin_1_1_plugin_exception_a64bddfa50fcebffb9b5db172a31f7911}


Return the error message. 

\begin{DoxyReturn}{Returns}
const char $\ast$ 
\end{DoxyReturn}


The documentation for this class was generated from the following files\+:\begin{DoxyCompactItemize}
\item 
include/\+Plugin/plugin\+\_\+exception.\+h\item 
src/\+Plugin/plugin\+\_\+exception.\+cc\end{DoxyCompactItemize}
